### ISSUE TITLE
Vercel ignore not-found.tsx on NextJS 13

### ISSUE THREAD

**Issue by shahargl** [vercel/next.js#50590]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Mon Aug 22 20:20:05 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T8101
    Binaries:
      Node: 19.6.0
      npm: 9.6.4
      Yarn: 1.22.19
      pnpm: 8.5.0
    Relevant packages:
      next: 13.4.3
      eslint-config-next: 13.4.3
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/keephq/keep/blob/main/keep-ui/app/not-found.tsx

### To Reproduce

The code is here - https://github.com/keephq/keep/tree/main/keep-ui
Specifically, `not-found.tsx` is here - https://github.com/keephq/keep/blob/main/keep-ui/app/not-found.tsx

### Describe the Bug

When running in localhost, the `not-found.tsx` works:
![image](https://github.com/vercel/next.js/assets/12069200/6c5654e2-3970-460f-a37e-a05f5965f64e)

However, when deploying to Vercel, I get the default 404 page:
![image](https://github.com/vercel/next.js/assets/12069200/cda1b600-40f2-4c51-9104-dee4e5db8cd5)

### Expected Behavior

Vercel should respect the custom `not-found.tsx`

### Which browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

**Comment by 4rweb** [comment]
I'm encountering a similar issue on my website. Vercel seems to be disregarding my 'not-found.tsx' file. You can see the problem by visiting https://dktsalu.vercel.app/skdhjs

**Comment by 4rweb** [comment]
If I put `404.tsx` in the folder `/pages/404.tsx` works! But on `/app/404.tsx` not working!

**Comment by icyJoseph** [comment]
No, that's not the way to go.

1. Delete pages/404.tsx
2. Create app/not-found.tsx

That's all there is to it. 

What next.js version do you have?

**Comment by 4rweb** [comment]
Using this version: "next": "13.4.4",

I create app/not-found.tsx, and I remove pages/404.tsx, but not working! 

Look: https://dktsalu-ax0ypkqnf-informaticadkt.vercel.app/testerror

For some reason that I don't know! Maybe it is the version of NextJS

**Comment by icyJoseph** [comment]
Wow, that's really strange. I updated to 13.4.4 and I get the 404 page, no problem.

Could there be an issue with your 404?

In your Vercel dashboard, for the https://dktsalu-ax0ypkqnf-informaticadkt.vercel.app/testerror deployment, can you check the Sources Tab? There should be a 404 file, what contents does it have?

![Screenshot 2023-06-02 at 21 45 27](https://github.com/vercel/next.js/assets/21013447/56d28ce4-798a-4e64-bf08-961f4e70b816)

**Comment by 4rweb** [comment]
I can't print it because it is too large, but I copy the text

look to files in the source tab:

```
_next
blog
icons
_error
a andalan
a andalan-day
andalan-day.rsc
andalan.rsc
blog
blog.rse
busca
busca.rsc
a conheca-outros-metodos-c
conheca-outros-metodos-c
contato
contato.rso
contraceptivos-de-uso-injet
contraceptivos-de-uso-injet
index
index.rse
pilula-de-emergencia
pilula-de-emergencia.rso
politica
politica.rse
termos-de-uso
termos-de-uso.rse
500
Andalan.png
android-chrome-192×192.png
android-chrome-512×512.png
apple-touch-icon.png
bg-404.ong
browsercontig.xmi
favicon-16×16.png
favicon-32x32.png
favicon.ico
logo.png
moon.svg
mstile-150×150.png
safari-pinned-tab.sg
site.webmanifest
sun.svg
```

I don't have 404 in the source!!!

**Comment by icyJoseph** [comment]
I think the issue might be that you have `pages/_errors` try deleting that one.

**Comment by 4rweb** [comment]
Hey @icyJoseph , thank you for your assistance! However, the solution you provided isn't working. I created a /page/404.tsx file, and it's working for https://www.dktsalu.com.br/skdfhjskd/.

The problem is that when the layout.tsx is located in the /app folder, the 404 page ignores the layout. Unfortunately, I need to update this website today, and I don't have much time.

Thank you again for your help!

**Comment by icyJoseph** [comment]
But I am not saying you should have a 404 page in pages. You must delete it! 

All you need is the app/not-found.tsx

What I think might be happening, is that if you also have pages/_error, which probably takes over app/not-found.tsx

So please no pages/404 and pages/_error. Remove both, but do not remove the app/not-found page, keep it!

**Comment by shahargl** [comment]
@icyJoseph I've renamed `not-found.tsx` to `under-constrcution.tsx` since it didn't work anyway.
This is the code before the rename: https://github.com/keephq/keep/blob/633bf32b6ba1cd5cc0b2320033f95fe727a14465/keep-ui/app/not-found.tsx

**Comment by icyJoseph** [comment]
Wow... Mmm there's quite a few moving parts on that repository.

I don't have access to my laptop this weekend. I wonder if you can try to test if the problem is the middleware?

**Comment by icyJoseph** [comment]
I wonder if the output standalone might be the problem? I haven't tried the not-found page with standalone activated.

**Comment by shahargl** [comment]
What do you mean by "standalone"? and If you want me to run some checks, let me know.

**Comment by icyJoseph** [comment]
You have standalone mode activated: https://github.com/keephq/keep/blob/main/keep-ui/next.config.js 

Also please take the next@13.4.4 update - hopefully tomorrow I get some time to help out and try different things, and see if we can fix it or create a reproduction repository of the issue

**Comment by icyJoseph** [comment]
@shahargl did you make any progress on this issue? Did you even test running `next build` and `next start` locally? AKA a production build, but locally?

**Comment by icyJoseph** [comment]
Yeah I tried your project locally and it worked locally, with and without standalone output... I wonder if the multi tenant thing is the issue but I'm not sure. Super suspicious... I'm out of ideas now

**Comment by clangerOg** [comment]
I'm encountering the same issue, which I described in this #50849 discussion. 

Looking at the created source output on Vercel it lloks like there actually is an `_error` and an `500` file, but I can't seem to find a `404` file. 

I could imagine that Next.js skips the generation of this file in some form because of some Vercel settings?
I will run a few more tests, but for now I am at a loss.

**Comment by ianbunag** [comment]
> I wonder if the output standalone might be the problem? I haven't tried the not-found page with standalone activated.

I also encountered this on:
- output `export`
- build script `next build`
- start script `npx serve@latest out`

In dev mode `next dev --turbo` the root not found page is being served, but on prod mode (build and start) the default 404 is shown.

Source code here https://gitlab.com/yvnbunag/scaffold/-/tree/main/src/templates/web/next-js/mtc (under construction).

**Comment by Gideonnn** [comment]
I have a similar experience as @ianbunag while having output set to `export`. Behavior is different when running a dev server compared to doing a production build.

One thing I noticed is that calling `notFound` from `next/navigation` does seem to route to the `not-found.tsx` page.

**Comment by Metarock** [comment]
Hiya, just wondering if you have any updates with this issue. I am having a similar experience but with `error.tsx`. Behavior in local environment works as expected however in vercel it is serving the default 500 error page rather than the `error.tsx`. I've opened up a discussion for it [#55130](https://github.com/vercel/next.js/discussions/55130#discussion-5604802).

**Comment by bernardogarza** [comment]
Any updates on this? I tried deleting the `pages` (I only had an `_Error` file in there, no `404`) folder completely but I'm still getting Vercel's 404 page instead of my `not-found`. As @Gideonnn mentioned the handling it with the `not-found()` function manually does fallbacks to the right page.
![image](https://github.com/vercel/next.js/assets/45184900/23ab64c1-4de6-4b60-9d6d-098fdaa76f37)

**Comment by Mihir867** [comment]
Hi, Anyone any updates on this ?
